### ISSUE TITLE
IsTrueForAny() incorrectly warns that it creates a new copy of the array without modifying the original

### ISSUE THREAD

**Issue by SlenderFox** [ItsDeltin/Overwatch-Script-To-Workshop#511]
IsTrueForAny() applies a boolean condition onto an array and returns true if any of the elements returned true. While true it does not modify the original, it also does not return a copy.

![image](https://github.com/user-attachments/assets/1db6e3b1-91e7-4817-96d3-0140b6f2f8f2)

**Comment by Syvul** [comment]
> While true it does not modify the original, it also does not return a copy.
> 
> ![image](https://private-user-images.githubusercontent.com/36467674/397914638-1db6e3b1-91e7-4817-96d3-0140b6f2f8f2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYwNjk3MzIsIm5iZiI6MTczNjA2OTQzMiwicGF0aCI6Ii8zNjQ2NzY3NC8zOTc5MTQ2MzgtMWRiNmUzYjEtOTFlNy00ODE3LTk2ZDMtMDE0MGI2ZjJmOGYyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA1VDA5MzAzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA1ODZhOGIxZDNhZGU2YzY5MjZlN2QzYjBlNjYwYzk0M2E3NWUyOWNkYzQxZmI2MjAxOGZlZmMwOTkwNjNiYTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.sWboanQdFXht-taRbmPTVKjHudXgkQFRmZJdpRNTClw)
Yes it evaluates on the copy internally, and leaves the original unchanged, IsTrueForAny as well as IsTrueForAll just return a boolean. Though it does this regardless of evaluating on true or false, the default return internally of these functions should be also false not true.
